pyproject.toml

```diff
@@ -42,7 +42,7 @@ pytest = "^7.1.2"
 pytest-xdist = "^3.0.2"
 pytest-rerunfailures = "^10.3"
 pytest-cov = "^4.0.0"
-ruff = "^0.0.202"
+ruff = "^0.3.4"
```

Comment:
This needed to be upgraded to work with the [Ruff VSCode Extension](https://marketplace.visualstudio.com/items?itemName=charliermarsh.ruff). AFAICT, the upgrade doesn't affect linting behavior, only how the configuration is specified.